shared/modules/doc/README.md

```diff
@@ -1,53 +1,66 @@
-## Implementation
+A module common interface is an interface that defines the methods that we modelled as common to multiple modules. For example: the ability to start/stop a module is pretty common and for that we defined the `InterruptableModule` interface.
+There are some interfaces that are common to multiple modules and we followed the [Interface segregation principle](https://en.wikipedia.org/wiki/Interface_segregation_principle) and also [Rob Pike's Go Proverb](https://youtu.be/PAAkCSZUG1c?t=317):
 
-In order to maintain code agnostic from the inception of the implementation, protobuf3 is utilized for all structures in this package.
+> The bigger the interface, the weaker the abstraction.
 
-It is important to note, that while Pocket V1 strives to not share objects between modules, the genesis module will inevitably overlap between other modules.
+These interfaces that can be embedded in modules are defined in `shared/modules/module.go`. GoDoc comments will provide you with more information about each interface.
 
-Another architecture worth considering and perhaps is more optimal as the project nears mainnet is allowing each module to create and maintain their own genesis object and config files
+### **Module**
 
-### Code Organization
+A module is an abstraction (go interface) of a self-contained unit of functionality that carries out a specific task/set of tasks with the idea of being reusable, modular, testable and having a clear and concise API. A module might implement 0..N common interfaces. You can find additional details in the [Modules in detail](#modules-in-detail) section below.
 
-```bash
-genesis
-├── docs
-│   ├── CHANGELOG.md    # Genesis module changelog
-│   ├── README.md       # Genesis module README
-├── proto
-│   ├── account.proto   # account structure
-│   ├── actor.proto     # actor structure
-│   ├── config.proto    # configuration structure
-│   ├── gov.proto       # params structure
-│   ├── state.proto     # genesis state structure
-├── test_artifacts      # the central point of all testing code (WIP)
-│   ├── generator.go    # generate the genesis and config.json for tests and build
-│   ├── gov.go          # default testing parameters
-```
+### **Module mock**
+
+A mock is a fake implementation of a module that is used for testing purposes. It is used to simulate the behaviour of the module and to verify that the module is interacting with other modules correctly. Mocks are generated using `go:generate` directives together with the `mockgen` tool.
+
+### **Base module**
 
-TODO(#235): Update once runtime configs are implemented
+A base module is a module that implements a common interface exposing the most basic logic. They are meant to be embedded in the structs of the modules that implement the common interface when there's no special logic to be implemented and improve DRYness (Don't Repeat Yourself) and to reduce boilerplate code. You can find the base modules in the `shared/modules/base_modules` package.
```

Comment:
Proposed changes:
1. Replacing the pronoun "they" in the second sentence with "base modules" because it ("they") would be referencing the noun phrase "base module" from the previous sentence which has a different plurality.
1. Split and rephrase end of second sentence.
1. Emboldening the word "embedded" and "and" to highlight the applicability of any given base module.

```suggestion
A base module is a module that implements a common interface, exposing the most basic logic. Base modules are meant to be **embedded** in module structs which implement this common interface **and** don't need to override the respective interface member(s). The intention being to improve DRYness (Don't Repeat Yourself) and to reduce boilerplate code. You can find the base modules in the `shared/modules/base_modules` package.
```

Reply:
Thank you for your patience @bryanchriswhite! Also for taking the time to explain the proposed change 🙏